### PR TITLE
Made account fetch as first step fro AMP (driven by modularity)

### PR DESCRIPTION
**Pull request by DGarbar**
This change will align logic with AuctionRequestFactory.

### CODE REVIEW

**Comment by DGarbar** [comment]
Notice test-appnexus-bid-request.json change. 
I don't see problem with sending this to the bidder.

File: src/main/java/org/prebid/server/auction/requestfactory/AmpRequestFactory.java [hunk 1]
```diff
@@ -134,22 +147,49 @@ public AmpRequestFactory(StoredRequestProcessor storedRequestProcessor,
...
-    private Future<Tuple2<BidRequest, List<String>>> createBidRequest(RoutingContext context) {
-        final String tagId = context.request().getParam(TAG_ID_REQUEST_PARAM);
-        if (StringUtils.isBlank(tagId)) {
-            return Future.failedFuture(new InvalidRequestException("AMP requests require an AMP tag_id"));
+    private Future<Tuple2<BidRequest, List<String>>> updateBidRequest(AuctionContext auctionContext) {
+        final BidRequest receivedBidRequest = auctionContext.getBidRequest();
+        final ExtRequest requestExt = receivedBidRequest != null ? receivedBidRequest.getExt() : null;
+        final ExtRequestPrebid prebid = requestExt != null ? requestExt.getPrebid() : null;
+        final ExtStoredRequest storedRequest = prebid != null ? prebid.getStoredrequest() : null;
+        final String storedRequestId = storedRequest != null ? storedRequest.getId() : null;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/auction/requestfactory/AmpRequestFactory.java:176]: Pls extract to separate method.

File: src/main/java/org/prebid/server/auction/requestfactory/AmpRequestFactory.java [hunk 2]
```diff
@@ -134,22 +147,49 @@ public AmpRequestFactory(StoredRequestProcessor storedRequestProcessor,
...
+
+        final Account account = auctionContext.getAccount();
+        final String accountId = account != null ? account.getId() : null;
+
+        if (StringUtils.isBlank(storedRequestId)) {
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/auction/requestfactory/AmpRequestFactory.java:181]: Move this upper right after `storedRequestId` resolving.

File: src/main/java/org/prebid/server/auction/StoredRequestProcessor.java [hunk 3]
```diff
@@ -134,8 +134,7 @@ public static StoredRequestProcessor create(long defaultTimeout,
     /**
      * Fetches AMP request from the source.
      */
-    public Future<BidRequest> processAmpRequest(String accountId, String ampRequestId) {
-        final BidRequest bidRequest = defaultBidRequest != null ? defaultBidRequest : BidRequest.builder().build();
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/auction/StoredRequestProcessor.java:138]: Where we're processing `defaultBidRequest` now?

**Review by rpanchyk — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/auction/StoredRequestProcessor.java (hunk 3 above)
Comment (DGarbar) [reply src/main/java/org/prebid/server/auction/StoredRequestProcessor.java:138]: We will merge defaultBidRequest later in workflow. I think previous code was using defaultBidRequest to align code with auction. Currently it even more aligned
```
    private BidRequest mergeDefaultRequest(BidRequest bidRequest) {
        return jsonMerger.merge(bidRequest, defaultBidRequest, BidRequest.class);
    }
```

**Review by rpanchyk — APPROVED** [review]

**Pull request merged**